### ISSUE TITLE
Unable to boot Sidekiq with 4.4.0

### ISSUE THREAD

**Issue by geoffharcourt** [getsentry/sentry-ruby#1431]
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Actual behavior**

**Environment**
 - Ruby Version: 2.7.3
 - SDK Version: 4.4.0
 - Integration Versions (if any):
  - e.g. Rails 6.1.3.1, Sidekiq 6.2.1, Sidekiq Pro 5.2.2, Sidekiq Enterprise 2.2.2

After upgrading to 4.4.0 this morning, our app's Sidekiq server stopped working. It looks there's an infinite loop of async exception notifications that fills up Redis until an OOM error. There might be a conflict between `sentry-ruby-core` and the New Relic RPM (we're using 6.15.0), the error appears to happen as the New Relic agent starts.

Our logs are filled with this over and over again:

```
2021-05-05T14:17:19.278004+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:51:in `block in request_with_newrelic_trace'
2021-05-05T14:17:19.278004+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent.rb:507:in `disable_all_tracing'
2021-05-05T14:17:19.278005+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:50:in `request_with_newrelic_trace'
2021-05-05T14:17:19.278005+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/sentry-ruby-core-4.4.0/lib/sentry/net/http.rb:9:in `request'
2021-05-05T14:17:19.278005+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:52:in `block (2 levels) in request_with_newrelic_trace'
2021-05-05T14:17:19.278006+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/tracer.rb:371:in `capture_segment_error'
2021-05-05T14:17:19.278006+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:51:in `block in request_with_newrelic_trace'
2021-05-05T14:17:19.278006+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent.rb:507:in `disable_all_tracing'
2021-05-05T14:17:19.278023+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:50:in `request_with_newrelic_trace'
2021-05-05T14:17:19.278024+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/sentry-ruby-core-4.4.0/lib/sentry/net/http.rb:9:in `request'
2021-05-05T14:17:19.278024+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:52:in `block (2 levels) in request_with_newrelic_trace'
2021-05-05T14:17:19.278024+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/tracer.rb:371:in `capture_segment_error'
2021-05-05T14:17:19.278025+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:51:in `block in request_with_newrelic_trace'
2021-05-05T14:17:19.278025+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent.rb:507:in `disable_all_tracing'
2021-05-05T14:17:19.278025+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:50:in `request_with_newrelic_trace'
2021-05-05T14:17:19.278026+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/sentry-ruby-core-4.4.0/lib/sentry/net/http.rb:9:in `request'
2021-05-05T14:17:19.278026+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:52:in `block (2 levels) in request_with_newrelic_trace'
2021-05-05T14:17:19.278028+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/tracer.rb:371:in `capture_segment_error'
2021-05-05T14:17:19.278029+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:51:in `block in request_with_newrelic_trace'
2021-05-05T14:17:19.278029+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent.rb:507:in `disable_all_tracing'
2021-05-05T14:17:19.278029+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:50:in `request_with_newrelic_trace'
2021-05-05T14:17:19.278030+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/sentry-ruby-core-4.4.0/lib/sentry/net/http.rb:9:in `request'
2021-05-05T14:17:19.278030+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:52:in `block (2 levels) in request_with_newrelic_trace'
2021-05-05T14:17:19.278030+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/tracer.rb:371:in `capture_segment_error'
2021-05-05T14:17:19.278031+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:51:in `block in request_with_newrelic_trace'
2021-05-05T14:17:19.278031+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent.rb:507:in `disable_all_tracing'
2021-05-05T14:17:19.278031+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:50:in `request_with_newrelic_trace'
2021-05-05T14:17:19.278033+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/sentry-ruby-core-4.4.0/lib/sentry/net/http.rb:9:in `request'
2021-05-05T14:17:19.278033+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:52:in `block (2 levels) in request_with_newrelic_trace'
2021-05-05T14:17:19.278033+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/tracer.rb:371:in `capture_segment_error'
2021-05-05T14:17:19.278034+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:51:in `block in request_with_newrelic_trace'
2021-05-05T14:17:19.278034+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent.rb:507:in `disable_all_tracing'
2021-05-05T14:17:19.278034+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:50:in `request_with_newrelic_trace'
2021-05-05T14:17:19.278035+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/sentry-ruby-core-4.4.0/lib/sentry/net/http.rb:9:in `request'
2021-05-05T14:17:19.278035+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:52:in `block (2 levels) in request_with_newrelic_trace'
2021-05-05T14:17:19.278035+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/tracer.rb:371:in `capture_segment_error'
2021-05-05T14:17:19.278036+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:51:in `block in request_with_newrelic_trace'
2021-05-05T14:17:19.278036+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent.rb:507:in `disable_all_tracing'
2021-05-05T14:17:19.278036+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/newrelic_rpm-6.15.0/lib/new_relic/agent/instrumentation/net.rb:50:in `request_with_newrelic_trace'
2021-05-05T14:17:19.278037+00:00 app[worker.1]: /app/vendor/bundle/ruby/2.7.0/gems/sentry-ruby-core-4.4.0/lib/sentry/net/http.rb:9:in `request'
```

**Comment by st0012** [comment]
@geoffharcourt can you try setting `prepend_net_instrumentation: true` in your `newrelic.yml`?

**Comment by geoffharcourt** [comment]
It looks like this resolved the issues, thanks.

**Comment by st0012** [comment]
@geoffharcourt I just released `sentry-ruby 4.4.1`, which can help you avoid the above change. sorry for the inconvenience.
